acvm/src/lib.rs

```diff
@@ -49,6 +50,12 @@ pub enum OpcodeResolutionError {
     IncorrectNumFunctionArguments(usize, BlackBoxFunc, usize),
 }
 
+#[derive(PartialEq, Eq, Debug)]
+pub enum GateResolution {
+    Resolved,                // Gate is solved
```

Comment:
I think we previously removed this enum, why is it being added again?

Reply:
Yes, @TomAFrench already suggested this. I will remove this part and create another PR.
For the record, I copied the non-recursive solved from a long-standing PR which included it, but I should not have include this refactoring.

Reply:
It's removed